superset-frontend/src/explore/components/controls/ContourControl/ContourPopoverControl.tsx

```diff
@@ -0,0 +1,311 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState, useEffect } from 'react';
+import { Row, Col } from 'src/components';
+import Button from 'src/components/Button';
+import Tabs from 'src/components/Tabs';
+import { legacyValidateInteger, styled, t } from '@superset-ui/core';
+import ControlHeader from '../../ControlHeader';
+import TextControl from '../TextControl';
+import ColorPickerControl from '../ColorPickerControl';
+import {
+  ContourPopoverControlProps,
+  colorType,
+  contourType,
+  errorMapType,
+} from './types';
+
+enum CONTOUR_TYPES {
+  ISOLINE = 'ISOLINE',
+  ISOBAND = 'ISOBAND',
+}
```

Comment:
nit: PascalCase tends to be used for enums, at least in the official TS docs: https://www.typescriptlang.org/docs/handbook/enums.html#string-enums

```suggestion
enum CONTOUR_TYPES {
  Isoline = 'ISOLINE',
  Isoband = 'ISOBAND',
}
```